### PR TITLE
feat: show convert modal when user uses older versions of spec

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Similar logic like in the [playground](https://playground.asyncapi.io/):

- Show modal when user will use older version (less than `2.X.X` version) - always show
- Show modal when user will use non latest version - only one time per day. Information about showing is stored in the sessionStorage, so it will be also showed when user will start new session in the browser.
- Add unit tests

image

**Related issue(s)**
See also #80

### CODE REVIEW

**Comment by magicmatatjahu** [comment]
@boyney123 Do you wanna make another review? 😅

File: src/services/specification.service.ts [hunk 1]
```diff
@@ -104,6 +107,29 @@ export class SpecificationService {
     return false;
   }
 
+  static shouldInformAboutLatestVersion(
+    version: string,
+  ): boolean {
+    const oneDay = 24 * 60 * 60 * 1000; /* ms */
```
Comment (jonaslagoni) [inline src/services/specification.service.ts:113]: Not sure what I feel about this 😄 In theory if you said no once, I think it should not ask again, but discretely show it somewhere, that the document is not up to date 🤔 Thoughts?

File: src/components/Modals/ConvertToLatestModal.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        <div className="mt-8">
+          <a
+            className="underline"
+            href={`https://www.asyncapi.com/blog/release-notes-${latestVersion}`}
+            target="_blank"
+            rel="nofollow noopener noreferrer"
+          >
+            See the release notes
+          </a>
+        </div>
```
Comment (jonaslagoni) [inline src/components/Modals/ConvertToLatestModal.tsx:72]: Should we not show the release notes for each version inbetween? i.e. if you have a document with version `2.0.0` and we currently are in `2.2.0` it should show release notes for `2.1.0` and `2.2.0` 🤔

File: src/services/specification.service.ts (hunk 1 above)
Comment (magicmatatjahu) [reply src/services/specification.service.ts:113]: I don't want to show it on every time, so it shows once a day (or once a session) - just like in playground. I can show information that there is a new version in the terminal, next to `Valid/Invalid` info, that the document does not use the latest version. What do you think?

Comment (jonaslagoni) [reply src/services/specification.service.ts:113]: The terminal part sounds like a good idea 👍 

I don't think it should re-ask each day though, once is enough, but that's just my preference 😄

File: src/components/Modals/ConvertToLatestModal.tsx (hunk 2 above)
Comment (magicmatatjahu) [reply src/components/Modals/ConvertToLatestModal.tsx:72]: It only shows release notes for latest, but it's a good idea to show all between current version and latest. I will propagate it. Thanks!

Comment (magicmatatjahu) [reply src/components/Modals/ConvertToLatestModal.tsx:72]: Mock for `2.0.0` version (and also for older than `2.0.0` - we don't have release notes for `2.0.0` so I won't show any link):

image

Mock for `2.1.0` version:

image

Comment (jonaslagoni) [reply src/components/Modals/ConvertToLatestModal.tsx:72]: This reminds me based on last PR - https://github.com/asyncapi/studio/pull/100

Should the user be able to select another version to convert to then the latest version? 🤔

File: src/services/specification.service.ts (hunk 1 above)
Comment (magicmatatjahu) [reply src/services/specification.service.ts:113]: image

File: src/components/Modals/ConvertToLatestModal.tsx (hunk 2 above)
Comment (magicmatatjahu) [reply src/components/Modals/ConvertToLatestModal.tsx:72]: I can do this :)

File: src/services/specification.service.ts (hunk 1 above)
Comment (jonaslagoni) [reply src/services/specification.service.ts:113]: Can you add a red negative sign, opposite of `valid` basically, and make it so you can click it to convert? 🤔

File: src/components/Modals/ConvertToLatestModal.tsx (hunk 2 above)
Comment (magicmatatjahu) [reply src/components/Modals/ConvertToLatestModal.tsx:72]: But... Between 2.0.0 and 2.2.0 there are only new things that help more than hinder. We could implement something like this when we will release 3.0.0, what do you think? I can't think of a use case if someone wanted a version smaller than 2.2.0 (atm).

File: src/services/specification.service.ts (hunk 1 above)
Comment (jonaslagoni) [reply src/services/specification.service.ts:113]: Just want the `NOT LATEST` to be more visible to the user 🙂

File: src/components/Modals/ConvertToLatestModal.tsx (hunk 2 above)
Comment (jonaslagoni) [reply src/components/Modals/ConvertToLatestModal.tsx:72]: It makes sense yea, I cant think of a use-case either, however when you have the option in the other functionality to decide between versions, I would assume the same behavior where consistent here as well 🙂

Comment (magicmatatjahu) [reply src/components/Modals/ConvertToLatestModal.tsx:72]: Ok, I will do it tomorrow :) Thanks for suggestions!

File: src/services/specification.service.ts (hunk 1 above)
Comment (magicmatatjahu) [reply src/services/specification.service.ts:113]: Ok, I will do it tomorrow :) Thanks for suggestions! `NOT LATEST` will have warning sign.

**Comment by magicmatatjahu** [comment]
@jonaslagoni Changes:

Info about not latest version in terminal:

image

After clicking in the `Not latest` button we will see modal to convert doc to latest version (without selection):

image

Modal that shows up to convert old (not supported) versions - cancel button is disabled:

image

Modal that shows up to convert old (but supported) versions:

image

**Review by jonaslagoni — APPROVED** [review]
LGTM 👍

**Comment by magicmatatjahu** [comment]
@boyney123 No worries :)

**Pull request merged**